src/Elements/types.jl

```diff
@@ -0,0 +1,180 @@
+const HEADERS   = ["#", "##", "###", "####", "#####", "######"]
+const EMPHASIS  = ["*", "**"]
+
+const CATEGORIES = [:module,    :function, :method, :macro, :type,
+                    :typealias, :global,   :symbol, :tuple, :aside]
+
+
+## Config Nodes
+abstract ConfigNode
+
+type Config <: ConfigNode
+    objfilter               :: Any
+    style_obj               :: ASCIIString
+
+    const defaults = Dict{Symbol, Any}(
+        :objfilter                  => CATEGORIES,
+        :style_obj                  => "####",
```

Comment:
> Or do you know how to allow arbitrary keyword arguments to a node kile this:

No, that can't be done and also would be a bit surprising to users.

Having keyword arguments at the end of a node doesn't seem unclear to me. ``config`` nodes should have higher precedence the further down the argument list they are found, thus the keyword arguments right at the end (after all ``config`` nodes) would have the highest precedence. Higher precedence would override lower precedence values.

> Propably this can be done - personally I do not see the advantage at the moment - so maybe leave it until the need arises?

It would remove the need for

```jl
const HEADERS   = ["#", "##", "###", "####", "#####", "######"]
const EMPHASIS  = ["*", "**"]
```

wouldn't it? ``HEADERS`` and ``EMPHASIS`` isn't that flexible if you'd like to style something a bit differently. Using a ``Function`` to style things is, as far as I can tell, a more general and simpler solution.

Granted, if the package author writes an incorrect styling function, then bad things could happen, but honestly that's just a matter of documenting it properly. (The ``Base.sort`` function takes an arbitrary ``lt = `` function for sorting.)